yt_dlp/extractor/theplatform.py

```diff
@@ -297,6 +302,30 @@ def _real_extract(self, url):
 
         formats, subtitles = self._extract_theplatform_smil(smil_url, video_id)
 
+        note = 'Downloading m3u8 information'
+        errnote = 'Failed to download m3u8 information'
+        # on at least one site reading just the SMIL data misses some formats as well as subtitles.
+        # Therefore also read the m3u8 file and add those formats and subtitles
+        m3u8_url = url.split("&")[0]
+        ext = urlhandle_detect_ext(self._request_webpage(
+            HEADRequest(m3u8_url),
+            video_id,
+            fatal=False))
+        if ext == 'm3u8':
+            res = self._download_webpage_handle(
+                m3u8_url, video_id,
+                note=note,
+                errnote=errnote,
+                fatal=False)
+            if res is not False:
+                m3u8_doc, urlh = res
+                formats2, subtitles2 = self._parse_m3u8_formats_and_subtitles(
+                    m3u8_doc, m3u8_url, entry_protocol='m3u8_native',
+                    note=note, errnote=errnote, fatal=False,
+                    video_id=video_id)
+                formats += formats2
+                subtitles.update(subtitles2)
+
```

Comment:
`update_url` can be imported from utils
```suggestion
        # With some sites, extracting from manifest URL directly yields more formats
        m3u8_url = update_url(url, query='mbr=true&manifest=m3u', fragment=None)
        ext = urlhandle_detect_ext(self._request_webpage(
            HEADRequest(m3u8_url), video_id, 'Checking for additional m3u8 formats', fatal=False))
        if ext == 'm3u8':
            m3u8_fmts, m3u8_subs = self._extract_m3u8_formats_and_subtitles(m3u8_url, video_id, fatal=False)
            formats.extend(m3u8_fmts)
            self._merge_subtitles(m3u8_subs, target=subtitles)

```

Reply:
Ah, OK, missed that. Will update once I get done with the tests thing.